cmd/maya-apiserver/cstor-operator/cspc/operations.go

```diff
@@ -108,22 +121,31 @@ func getRaidGroupType(group apis.RaidGroup, poolSpec *apis.PoolSpec) string {
 }
 
 // addBlockDeviceToGroup adds block devices to the provided raid group on CSP
-func addBlockDeviceToGroup(group *apis.RaidGroup, csp *apis.NewTestCStorPool) *apis.NewTestCStorPool {
+func (pc *PoolConfig) addBlockDeviceToGroup(group *apis.RaidGroup, csp *apis.NewTestCStorPool) *apis.NewTestCStorPool {
 	for i, groupOnCSP := range csp.Spec.RaidGroups {
 		groupOnCSP := groupOnCSP
 		if isRaidGroupPresentOnCSP(group, csp) {
 			if len(group.BlockDevices) > len(groupOnCSP.BlockDevices) {
 				newBDs, err := getAddedBlockDevicesInGroup(group, &groupOnCSP)
 				if err != nil {
-					glog.V(2).Infof("Failed to get newly added block device on group {%+v}", group)
+					glog.V(2).Infof("Failed to get newly "+
+						"added block device on group {%+v}", group)
 				}
 				if len(newBDs) == 0 {
-					glog.V(2).Infof("No new block devices added for group {%+v} on csp %s", groupOnCSP, csp.Name)
+					glog.V(2).Infof("No new block devices "+
+						"added for group {%+v} on csp %s", groupOnCSP, csp.Name)
 				}
 				for _, bdName := range newBDs {
-					csp.Spec.RaidGroups[i].BlockDevices = append(csp.Spec.RaidGroups[i].BlockDevices, apis.CStorPoolClusterBlockDevice{BlockDeviceName: bdName})
+					err := pc.isBDUsable(bdName)
+					if err != nil {
+						glog.Errorf("could not use bd %s for "+
+							"expanding pool %s:%s", bdName, csp.Name, err.Error())
+						break
+					}
+					csp.Spec.RaidGroups[i].BlockDevices =
```

Comment:
can we maintain the same order of raidGroups and BDs between cspc and csp?

Reply:
We cannot depend on that definitely because somebody can add raid groups in between in a CSPC.

Reply:
I understand we can't depend on this.. ok.. lets avoid this